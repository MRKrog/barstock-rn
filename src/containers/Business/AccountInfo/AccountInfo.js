import React, { Component } from "react";
import { View, Text } from "react-native";
import styles from "./AccountInfo.style";
import Footer from "../../../components/Footer/Footer";
import { connect } from "react-redux"

export class AccountInfo extends Component{
  render() {
    const info = this.props.businessInfo.business.data.attributes
    return (
      <View style={styles.conatiner}>
        <View style={styles.info_container}>
          <View style={styles.account_nameContainer}>
            <Text style={styles.account_name}>Account Information</Text>
          </View>
          <View style={styles.info_TextContainer}>
            <View style={styles.info_section}>
            <Text style={styles.info_title}>
                  Personal
                </Text>
                <Text style={styles.info_text}>
                  Name: {info.name}
                </Text>
                <Text style={styles.info_text}>
                  Address: {info.address}
                </Text>
                <Text style={styles.info_text}>
                  Email: {info.email}
                </Text>
                <Text style={styles.info_text}>
                  Phone: {info.phone_number}
                </Text>
            </View>
            <View style={styles.info_section}>
              <Text style={styles.info_title}>
                Representative
              </Text>
              <Text style={styles.info_text}>
                  Name: {info.representative.rep_name}
                </Text>
                <Text style={styles.info_text}>
                  Phone: {info.representative.rep_phone_number}
                </Text>
                <Text style={styles.info_text}>
                  Email: {info.representative.rep_email}
                </Text>
            </View>
            <View style={styles.info_section}>
              <Text style={styles.info_title}>
                {info.distributor.distributor_name}
              </Text>
              <Text style={styles.info_text}>
                  Address: {info.distributor.distributor_address}
              </Text>
            </View>
          </View>
        </View>
        <Footer />
      </View>
    )
  }
}

export const mapStateToProps = (state) => ({
  businessInfo: state.businessInfo
})

export default connect(mapStateToProps)(AccountInfo)
