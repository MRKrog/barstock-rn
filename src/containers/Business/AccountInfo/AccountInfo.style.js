import { Dimensions } from "react-native";

const screenWidth = Dimensions.get("window").width
const screenHeight = Dimensions.get("window").height

export default {
  conatiner: {
    flex: 1,
    display: "flex",
    justifyContent: "space-between"
  },
  info_container: {
    display: "flex"
  },
  account_nameContainer: {
    backgroundColor: "#2c4969",
    padding: 10,
    shadowOpacity: 0.3,
    shadowRadius: 3,
    shadowColor: '#231f20',
    shadowOffset: { height: 4, width: 0 },
  },
  account_name: {
    fontSize: 22,
    textTransform: "uppercase",
    color: "#ffffff",
    textAlign: "center",
    fontFamily: 'abel'
  },
  info_TextContainer: {
    backgroundColor: "#ffffff",
    borderRadius: 5,
    shadowOpacity: 0.3,
    shadowRadius: 3,
    shadowColor: '#231f20',
    shadowOffset: { height: 4, width: 0 },
  },
  info_section: {
    display: "flex",
    flexDirection: "column",
    margin: 10,
    borderBottomColor: "#464646",
    borderBottomWidth: 1
  },
  info_text: {
    margin: 5,
    fontSize: 20,
    fontFamily: "Lato-Light",
    color: "#464646",
    paddingBottom: 5
  },
  info_title: {
    margin: 5,
    fontSize: 20,
    fontFamily: "Lato-Regular",
    color: "#464646",
    paddingBottom: 5,
    textDecorationLine: "underline"
  }
}
