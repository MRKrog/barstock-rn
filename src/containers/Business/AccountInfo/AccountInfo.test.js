import { shallow } from "enzyme"
import { AccountInfo } from "./AccountInfo"
import React from "react"

describe("AccountInfo", () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <AccountInfo />
    )
  })

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })
})