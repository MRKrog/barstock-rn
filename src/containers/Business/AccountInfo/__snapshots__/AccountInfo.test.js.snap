// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountInfo should match snapshot 1`] = `undefined`;
